### PR TITLE
docs: Add siderail for content pages and improve layouts [Job 111086]

### PR DESCRIPTION
**Pull request by Aiden-Brine**
## Changes

- Adds a `BaseView` component that contains the styles that are shared between `ComponentView` and `ContentView`
- Related to this the page layout on content pages should now be consistent with component pages
- Content pages now have a siderail with anchor tags to jump to parts of the page
- If the siderail has many links (like on changelog pages) the siderail has a scrollbar independent of the main content scrollbar
- Some other cleanup and refactoring

## Testing

- Click around content and component pages and compare the two. The layouts should be matching
- The content pages should now have anchor links in the siderail. That being said as [Taylor recently pointed out](https://getjobber.slack.com/archives/C081WMPQLUV/p1734460480376659) there are issues with many of the anchor links in the site. So testing on this is more about confirming that the links are appearing (on most pages) and when the links are correct the styles and behaviour are as expected. Fixing all of the links will come in a later PR
- Go to a changelog page and ensure the behaviour when there are a huge number of links in the siderail is a good user experience

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/site/src/components/AnchorLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  readonly key: string;
+
+  /**
+   * An additional action to perform along with scrolling to the selected anchor
+   */
+  readonly additionalOnClickAction?: () => void;
+}
+
+export function AnchorLinks({
```
Comment (Aiden-Brine) [inline packages/site/src/components/AnchorLinks.tsx:21]: This is all functionality that was pulled from `ComponentView`

File: packages/site/src/components/ComponentLinks.tsx [hunk 2]
```diff
@@ -9,130 +9,87 @@ import { useAtlantisSite } from "../providers/AtlantisSiteProvider";
...
-        width: "200px",
-        padding: "36px var(--space-base)",
-        display: "flex",
-        flexDirection: "column",
-        flexShrink: "0",
-        boxSizing: "border-box",
-      }}
-    >
-      <Content spacing={"larger"}>
+    <Content spacing={"larger"}>
```
Comment (Aiden-Brine) [inline packages/site/src/components/ComponentLinks.tsx:32]: The diff looks much more involved than what was actually changed. This was essentially just pulling the anchor link functionality out into its own component

File: packages/site/src/components/ContentLoader.tsx [hunk 3]
```diff
@@ -28,11 +28,5 @@ export const ContentLoader = () => {
 
   const content = contentMap[type][name];
 
-  return (
-    <ContentView
-      intro={content.intro}
```
Comment (Aiden-Brine) [inline packages/site/src/components/ContentLoader.tsx:33]: intro and title were not being used. We should consider removing these from the mappings unless we plan to use them later?

File: packages/site/src/layout/BaseView.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+BaseView.Main = function Main({
+  children,
+}: {
+  readonly children: React.ReactNode;
+}) {
+  return (
+    <main
+      style={{
```
Comment (Aiden-Brine) [inline packages/site/src/layout/BaseView.tsx:15]: Styles were pulled from `ComponentView` for the most part

File: packages/site/src/layout/ComponentView.tsx [hunk 5]
```diff
@@ -186,55 +186,41 @@ export const ComponentView = () => {
...
-          overflowY: "scroll",
-          backgroundColor: "var(--color-surface)",
-          boxShadow: "var(--shadow-low)",
-          flexGrow: 1,
-        }}
-      >
-        <Box alignItems="center">
-          <div style={{ width: "100%", maxWidth: "768px" }}>
-            <Page width="narrow" title={PageMeta.title}>
-              <PageWrapper>
```
Comment (Aiden-Brine) [inline packages/site/src/layout/ComponentView.tsx:201]: As far as I can tell `PageWrapper` was providing no benefit here

File: packages/site/src/layout/ContentView.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Content } from "@jobber/components";
```
Comment (Aiden-Brine) [inline packages/site/src/layout/ContentView.tsx:1]: I moved `ContentView` into layouts since I think it belongs beside `ComponentView` and `BaseView`

Comment (taylorvnoj) [reply packages/site/src/layout/ContentView.tsx:1]: Great decision!

File: packages/site/src/layout/BaseView.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { Box } from "@jobber/components";
+import React from "react";
+
+export function BaseView({ children }: { readonly children: React.ReactNode }) {
```
Comment (taylorvnoj) [inline packages/site/src/layout/BaseView.tsx:4]: Love that you created this component!

This is a nit, not blocking, but we can make use of `PropsWithChildren` to simplify the type definitions. So we can have this instead:

```
export function BaseView({ children }: PropsWithChildren<{}>) {
  return <div style={{ display: "flex", height: "100dvh" }}>{children}</div>;
}
```

It's a utility type that automatically includes `children` as type React.ReactNode. Again, not a blocker, just something I learned a while ago.

File: packages/site/src/components/ContentLoader.tsx (hunk 3 above)
Comment (taylorvnoj) [reply packages/site/src/components/ContentLoader.tsx:39]: Interesting - I didn't realize we don't need them in the maps. But good call on using title for the tabs

**Review by taylorvnoj — APPROVED** [review]
I have one nit but it's not blocking - love the refactor and benefits that come with it (like some hooks pages now have 100% vh)

File: packages/site/src/components/ContentLoader.tsx (hunk 3 above)
Comment (Aiden-Brine) [reply packages/site/src/components/ContentLoader.tsx:39]: It seems like something we thought we would need but at the moment it doesn't seem like it we are. Something we could get rid of at the end when we are doing other cleanup

File: packages/site/src/layout/BaseView.tsx (hunk 7 above)
Comment (Aiden-Brine) [reply packages/site/src/layout/BaseView.tsx]: Ah good call, I have seen that in the code but forgot to reach for it when building this. Definitely an improvement and now added

**Pull request merged**